### PR TITLE
kafka(ticdc): close the async write asynchronously to prevent block the cdc server

### LINKED ISSUE #8578: async close the async write to prevent block the cdc server

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #8578

### What is changed and how it works?

* close the async writer asynchronously
* set `MaxAttemptes` to 2 and `WriteBackoffMin` to 10ms.

#### Tests 

 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/sink/kafka/v2/admin.go [hunk 1]
```diff
@@ -230,6 +230,9 @@ func (a *admin) CreateTopic(
 }
 
 func (a *admin) Close() {
```
Comment (overvenus) [inline pkg/sink/kafka/v2/admin.go:232]: Looks like there is no IO block in the `Close`.

Comment (3AceShowHand) [reply pkg/sink/kafka/v2/admin.go:232]: https://github.com/segmentio/kafka-go/blob/main/writer.go#L1016

When close the writer, it tries to flush all queued batches, and retry for `MaxAttempts`, this is time-consuming.

After setting the `MaxAttempts` to 2 and `WriteBackOffMin` to `10ms`, still need around 5 ~ 15s.

File: pkg/sink/kafka/v2/factory.go [hunk 2]
```diff
@@ -162,9 +162,13 @@ func (f *factory) newWriter(async bool) *kafka.Writer {
 		Transport:    f.transport,
 		ReadTimeout:  f.options.ReadTimeout,
 		WriteTimeout: f.options.WriteTimeout,
-		RequiredAcks: kafka.RequiredAcks(f.options.RequiredAcks),
-		BatchBytes:   int64(f.options.MaxMessageBytes),
-		Async:        async,
+		// For kafka cluster with a bad network condition,
+		// do not waster too much time to prevent long time blocking.
```
Comment (overvenus) [inline pkg/sink/kafka/v2/factory.go:166]: waste

**Review by overvenus — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**